pkgs/misc/vscode-extensions/update_installed_exts.sh

```diff
@@ -0,0 +1,67 @@
+#! /usr/bin/env nix-shell
+#! nix-shell -i bash -p curl jq unzip
+set -eu -o pipefail
+
+# Helper to just fail with a message and non-zero exit code.
+function fail() {
+    echo "$1" >&2
+    exit 1
+}
+
+# Helper to clean up after ourself if we're killed by SIGINT
+function clean_up() {
+    TDIR="${TMPDIR:-/tmp}"
+    echo "Script killed, cleaning up tmpdirs: $TDIR/vscode_exts_*" >&2
+    rm -Rf "$TDIR/vscode_exts_*"
+}
+
+function get_vsixpkg() {
+    N="$1.$2"
+
+    # Create a tempdir for the extension download
+    EXTTMP=$(mktemp -d -t vscode_exts_XXXXXXXX)
```

Comment:
If the script is aborted this, it would leave a temporary directory.
You can install a error handler to cleanup the directory when the function returns

```
EXTTMP=$(mktemp -d -t vscode_exts_XXXXXXXX)
trap "rm -rf $EXTTMP; trap - RETURN EXIT" RETURN EXIT
```

Reply:
Just saw `clean_up`. Both ways work. Also the local trap is more precise.